### ISSUE TITLE
Compatibility with io.quarkus:quarkus-rest

### ISSUE THREAD

**Issue by demtnt** [AkaZver/mapstruct-plugin#20]
Hi. There is a problem using this plugin with quarkus extension 'io.quarkus:quarkus-rest' after version 3.12.0. 

```
./gradlew clean --stacktrace

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'merit-integration-service'.
> java.util.ConcurrentModificationException (no error message)

* Try:
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

* Exception is:
org.gradle.api.ProjectConfigurationException: A problem occurred configuring root project 'merit-integration-service'.
        at org.gradle.configuration.project.LifecycleProjectEvaluator.wrapException(LifecycleProjectEvaluator.java:84)
        at org.gradle.configuration.project.LifecycleProjectEvaluator.addConfigurationFailure(LifecycleProjectEvaluator.java:77)
        at org.gradle.configuration.project.LifecycleProjectEvaluator.access$500(LifecycleProjectEvaluator.java:55)
        ....
Caused by: java.util.ConcurrentModificationException
        at org.gradle.api.internal.DefaultDomainObjectCollection$IteratorImpl.next(DefaultDomainObjectCollection.java:467)
        at com.github.akazver.gradle.plugins.mapstruct.MapstructPlugin.addOptionalDependencies(MapstructPlugin.java:55)
        at com.github.akazver.gradle.plugins.mapstruct.MapstructPlugin.lambda$apply$0(MapstructPlugin.java:36)
        at org.gradle.internal.code.DefaultUserCodeApplicationContext$CurrentApplication$1.execute(DefaultUserCodeApplicationContext.java:124)
        at org.gradle.configuration.internal.DefaultListenerBuildOperationDecorator$BuildOperationEmittingAction$1.run(DefaultListenerBuildOperationDecorator.java:173)
        at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)        
        
```
build.gradle.kts:
```
plugins {
    java
    id("io.quarkus")
    id("com.github.akazver.mapstruct") version "1.0.6"
}

repositories {
    mavenCentral()
    mavenLocal()
}

dependencies {
    implementation(enforcedPlatform("io.quarkus.platform:quarkus-bom:3.13.2"))
    implementation("io.quarkus:quarkus-rest") // without this line './gradlew clean' works
    implementation("io.quarkus:quarkus-arc")

}

group = "aaa.bbb"
version = "1.0-SNAPSHOT"

java {
    sourceCompatibility = JavaVersion.VERSION_21
    targetCompatibility = JavaVersion.VERSION_21
}

tasks.withType<JavaCompile> {
    options.encoding = "UTF-8"
    options.compilerArgs.add("-parameters")
}
```

P.S. There is problem with 'io.quarkus:quarkus-resteasy-reactive-jackson' and 'io.quarkus:quarkus-jdbc-postgresql' also

**Comment by AkaZver** [comment]
Hello, @demtnt !
Thank you for reporting

I've reproduced this behavior with the following build.gradle.kts
<details>
  <summary>script</summary>

```kotlin
plugins {
    java
    id("io.quarkus") version "3.13.2"
    id("com.github.akazver.mapstruct") version "1.0.6"
}

group = "com.example"
version = "1.0.0-SNAPSHOT"

repositories {
    mavenCentral()
    mavenLocal()
}

dependencies {
    implementation(enforcedPlatform("io.quarkus.platform:quarkus-bom:3.13.2"))
    implementation("io.quarkus:quarkus-rest")
    implementation("io.quarkus:quarkus-arc")
    implementation("io.quarkus:quarkus-resteasy-reactive-jackson")
    implementation("io.quarkus:quarkus-jdbc-postgresql")
}

java {
    sourceCompatibility = JavaVersion.VERSION_21
    targetCompatibility = JavaVersion.VERSION_21
}

tasks.withType<JavaCompile> {
    options.encoding = "UTF-8"
    options.compilerArgs.add("-parameters")
}
```

</details>

Java: Liberica JDK 21
Gradle: 8.10

Please, check the new version [1.0.8](https://plugins.gradle.org/plugin/com.github.akazver.mapstruct) with fix from PR #21

**Comment by demtnt** [comment]
1.0.8 works great.

Большое спасибо за оперативность.
